homeassistant/components/plugwise/config_flow.py

```diff
@@ -0,0 +1,83 @@
+"""Config flow for Plugwise integration."""
+import logging
+from typing import Any, Dict
+
+from Plugwise_Smile.Smile import Smile
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def _get_config_schema(input_dict: Dict[str, Any] = None) -> vol.Schema:
+    """
+    Return schema defaults for init step based on user input/config dict.
+
+    Retain info already provided for future form views by setting them as defaults in schema.
+    """
+    if input_dict is None:
+        input_dict = {}
+
+    return vol.Schema({vol.Required("host"): str, vol.Required("password"): str})
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """
+    Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+    websession = async_get_clientsession(hass, verify_ssl=False)
+    api = Smile(
+        host=data["host"], password=data["password"], timeout=30, websession=websession
+    )
+
```

Comment:
`InvalidAuth` is never thrown here

Reply:
Added accordingly - had a learning curve (and/or fight) but finally mocked it. Note as indicated on other comment we added InvalidAuthorization to the python module as well to make sure it handles accordingly.